kratos/processes/find_nodal_h_process.cpp

```diff
@@ -35,31 +35,42 @@ void FindNodalHProcess<THistorical>::Execute()
     }
 
     // Initialize NODAL_H values
-    IndexPartition<std::size_t>(mrModelPart.Nodes().size()).for_each([&](std::size_t index){
-        SetInitialValue(mrModelPart.NodesBegin() + index);
-    });
+    const double max = std::numeric_limits<double>::max();
+    if constexpr(THistorical) {
+        VariableUtils().SetVariable(NODAL_H, max, mrModelPart.Nodes());
+    } else {
+        VariableUtils().SetNonHistoricalVariable(NODAL_H, max, mrModelPart.Nodes());
+    }
 
     // Calculate the NODAL_H values
-    for(IndexType i=0; i < mrModelPart.Elements().size(); ++i) {
-        auto it_element = mrModelPart.ElementsBegin() + i;
-        auto& r_geom = it_element->GetGeometry();
+    for(auto& r_element : mrModelPart.Elements()) {
+        auto& r_geom = r_element.GetGeometry();
         const SizeType number_of_nodes = r_geom.size();
 
         for(IndexType k = 0; k < number_of_nodes-1; ++k) {
             double& r_h1 = GetHValue(r_geom[k]);
             for(IndexType l=k+1; l < number_of_nodes; ++l) {
-                double hedge = norm_2(r_geom[l].Coordinates() - r_geom[k].Coordinates());
+                const double hedge = norm_2(r_geom[l].Coordinates() - r_geom[k].Coordinates());
                 double& r_h2 = GetHValue(r_geom[l]);
 
                 // Get minimum between the existent value and the considered edge length
-                SetHValue(r_geom[k], std::min(r_h1, hedge));
-                SetHValue(r_geom[l], std::min(r_h2, hedge));
+                if constexpr(THistorical) {
+                    r_geom[k].FastGetSolutionStepValue(NODAL_H) = std::min(r_h1, hedge);
+                    r_geom[l].FastGetSolutionStepValue(NODAL_H) = std::min(r_h2, hedge);
+                } else {
+                    r_geom[k].SetValue(NODAL_H, std::min(r_h1, hedge));
+                    r_geom[l].SetValue(NODAL_H, std::min(r_h2, hedge));
```

Comment:
```suggestion
                    r_geom[k].GetValue(NODAL_H) = std::min(r_h1, hedge);
                    r_geom[l].GetValue(NODAL_H) = std::min(r_h2, hedge);
```
As it is already allocated in the above initialization.